diff -ru freetype.old/include/freetype/config/ftoption.h freetype/include/freetype/config/ftoption.h
--- freetype.old/include/freetype/config/ftoption.h	2017-08-23 05:55:39.000000000 +0100
+++ freetype/include/freetype/config/ftoption.h	2018-06-28 01:08:07.522264480 +0100
@@ -123,7 +123,7 @@
    * When this macro is not defined, FreeType offers alternative LCD
    * rendering technology that produces excellent output.
    */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /**************************************************************************
