diff -ru freetype.old/include/freetype/config/ftoption.h freetype/include/freetype/config/ftoption.h
--- freetype.old/include/freetype/config/ftoption.h	2017-08-23 05:55:39.000000000 +0100
+++ freetype/include/freetype/config/ftoption.h	2018-06-28 01:08:07.522264480 +0100
@@ -119,7 +119,7 @@
   /* rendering technology that produces excellent output without LCD       */
   /* filtering.                                                            */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
